### PR TITLE
S2U-46: New features

### PR DESCRIPTION
**Pull request by frasese**
- Added start and end synchronization dates. This will affect the "Run synchronization" process.
- In "New Site" hook, now we can filter sites by site property.
- Synchronization Job now has some configuration options
- New hook "Site Unpublished": Will remove all users from related Teams when a site is unpublished

This is paired with https://github.com/sakaiproject/sakai-reference/pull/210

### CODE REVIEW

File: microsoft-integration/admin-tool/src/main/webapp/WEB-INF/templates/fragments/siteFilter.html [hunk 1]
```diff
@@ -2,39 +2,40 @@
...
-	<th:block th:fragment="siteFilter" th:with="aux=${columnClassName} ?: 'col-md-4'">
+	<th:block th:fragment="siteFilter" th:with="aux=${columnClassName} ?: 'col-xs-4'">
 		<div class="row form-group">
 			<label th:classappend="${aux}" class="text-right" th:text="#{filter.site_type}"></label>
-			<div class="col-md-8">
+			<div class="col-xs-8" th:with="siteType=${prefix} ? ${__${prefix}__.siteType} : ${filter_siteType}">
 				<div>
 					<label>
-						<input th:name="${prefix} ? |${prefix}.siteType| : 'siteType'" type="radio" value="course" th:checked="${filter_siteType} == 'course'" />
+						<input th:name="${prefix} ? |${prefix}.siteType| : 'siteType'" type="radio" value="course" th:checked="${siteType}  == 'course'" />
```
[the remark this answers is not in the document]
Comment (frasese) [reply microsoft-integration/admin-tool/src/main/webapp/WEB-INF/templates/fragments/siteFilter.html:11]: dynamic nae given by thymeleaf

File: microsoft-integration/admin-tool/src/main/webapp/WEB-INF/templates/fragments/siteFilter.html [hunk 2]
```diff
@@ -2,39 +2,40 @@
...
 						<span th:text="#{filter.site_type.course}"></span>
 					</label>
 				</div>
 				<div>
 					<label>
-						<input th:name="${prefix} ? |${prefix}.siteType| : 'siteType'" type="radio" value="project" th:checked="${filter_siteType} == 'project'" />
+						<input th:name="${prefix} ? |${prefix}.siteType| : 'siteType'" type="radio" value="project" th:checked="${siteType} == 'project'" />
```
[the remark this answers is not in the document]
Comment (frasese) [reply microsoft-integration/admin-tool/src/main/webapp/WEB-INF/templates/fragments/siteFilter.html:17]: dynamic name given by thymeleaf

**Comment by frasese** [comment]
Maven tests are failing due to mailsender errors in not related tools.

**Comment by mpellicer** [comment]
We need a successfull CI build, I'll re-launch it.

**Review by bgarciaentornos — APPROVED** [review]
Looks nice, great work!

**Review by mpellicer — APPROVED** [review]

**Pull request merged**